### PR TITLE
remove unnecessary views from wmagent wmstats view

### LINKED ISSUE #4526: change wmstats agent side deploy script to only push necessary view

### PR DESCRIPTION
**Pull request by ticoann**
fixes #4526. @amaltaro, Alan please review. (Don't merge, need to be tested)

### CODE REVIEW

**Comment by amaltaro** [comment]
Just to make sure I understood:
 - WMStatsAgent contains wmstats views required only by the agent
 - WMStats contains wmstats views only required by WMStats.

I noticed the new WMStatsAgent files (updates, views, etc) are the same as those under WMStats. Shouldn't we then remove the same files under WMStats?

**Comment by ticoann** [comment]
Most updates to wmstats happens in WMAgent. And most views are used from central WMStats.
Problem we have was the views, although most views (except one 'time' view) was pushed to wmagent deployment which cause a lot of disk space. Current patch kept only one view for agent deployment.
The problem is it makes the duplicate code base for maintenance.  We could do something in deployment script. (while deploying the wmagent delete the files unused instead of having the duplicate code base) - one problem with this is it is not very obvious that which code are used for WMAgent unless you look at the deployment script.  - If you prefer that way, I can change the code.
Updates codes are kept for central wmstats although some of them are not used. For the following reason.
  - It allows separate script for manipulating the data. This is needed for some recovery script.
  - It doesn't contribute to increase of the data size

File: src/couchapps/WMStatsAgent/validate_doc_update.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   var isGlobalAdm = (userCtx.name === null)
+                     || matchesRole("admin","")
+                     || matchesRole("_admin","")
+                     || matchesRole("-admin","group:couchdb");
+
+   //---------------------------------
+   // Authorization rules for Myapp DB
+
+   // The following rule aplies for all operation types
+   var allowed = isGlobalAdm || matchesRole("production-operator","group:dataops")
```
Comment (amaltaro) [inline src/couchapps/WMStatsAgent/validate_doc_update.js]: Just trying to simplify, do we need to validate the user roles for agent updates? Couch is accessed via localhost with no certificate, maybe we could drop this and the one above.

**Comment by amaltaro** [comment]
I agree with you, IMO it's better to have a bit more code to maintain and have these things clearer.

File: src/couchapps/WMStatsAgent/validate_doc_update.js (hunk 1 above)
Comment (ticoann) [inline src/couchapps/WMStatsAgent/validate_doc_update.js]: You are right, I updated the permissions.

**Pull request merged**